lib/game_collection.rb

```diff
@@ -67,4 +67,31 @@ def find_percentage_of_ties
     tied_games = @games.count {|game| game.tie_game?}.to_f
     (tied_games/total_games).round(2)
   end
+
+  def goals_scored_per_team
+    team_hash = Hash.new()
+    @games.each do |game|
+      team_hash[game.home_team_id] ||= Hash.new(0)
+      team_hash[game.home_team_id][:my_goals] += game.home_goals
+      team_hash[game.home_team_id][:their_goals] += game.away_goals
+
+      team_hash[game.away_team_id] ||= Hash.new(0)
+      team_hash[game.away_team_id][:my_goals] += game.away_goals
+      team_hash[game.away_team_id][:their_goals] += game.home_goals
+    end
+    team_hash
+  end
+
+  def best_defense_pair
```

Comment:
Definitely the right idea here with the min_by. We want to compare averages here - so we need goals scored against a certain team / total games that team played.